autofix/index.js

```diff
@@ -0,0 +1,71 @@
+const gonzales = require('gonzales-pe');
+const fs = require('fs');
+const path = require('path');
+const walkSync = require('walk-sync');
+const yargs = require('yargs');
+const rulesMethods = require('./rules');
+
+const { log } = console;
+const argv = yargs.array('dir').array('rules').argv;
+let { rules, dir } = argv;
+
+// process user inputs
+if (rules) {
+  rules = rules.map(rule => rule.replace(/-([a-z])/g, (m, w) => w.toUpperCase())); // camelCased
+} else {
+  rules = Object.keys(rulesMethods);
+}
+
+// configuration
+const root = process.cwd();
+
+// prepare lookup
+const fixDirectories = dir;
+const scssFiles = fixDirectories.reduce((currentFiles, directory) => {
+  const directoryPath = path.join(root, directory);
+  const directoryFiles = walkSync(directoryPath)
+    .filter(file => path.extname(file) === '.scss')
+    .map(file => path.join(directory, file));
+
+  currentFiles.push(...directoryFiles);
+
+  return currentFiles;
+}, []);
+
+// define statistics rule map
+const fixStatistics = {};
+rules.forEach(rule => {
+  fixStatistics[rule] = 0;
+});
+
+// process fixing for each file
+scssFiles.forEach(file => {
+  const fullPath = path.join(root, file);
+  const contents = fs.readFileSync(fullPath, { encoding: 'utf8' });
+
+  if (contents.length == 0) {
+    return;
+  }
+
+  const parsedTree = gonzales.parse(contents, { syntax: 'scss' });
+
+  // for each rule, call with current file contents
+  rules.forEach(rule => {
+    const fixedCount = rulesMethods[rule](parsedTree);
+    fixStatistics[rule] += fixedCount;
+  });
+
+  // write fixed content to file
+  if (argv.ns !== true) {
```

Comment:
It's `no-save` options. I haven't found way to use full `--no-save` param, but I will be able to comment this properly with `yargs`